src/MultiFormats/MultiHash.cs

```diff
@@ -0,0 +1,570 @@
+        /// </summary>
+        /// <param name="s">
+        ///   A <see cref="Base58"/> encoded <b>MultiHash</b>.
+        /// </param>
+        /// <returns>
+        ///   A new <see cref="MultiHash"/>.
+        /// </returns>
+        /// <remarks>
+        ///    Equivalent to <code>new MultiHash(s)</code>
+        /// </remarks>
+        public static implicit operator MultiHash(string s) { return new MultiHash(s); }
+
+        /// <summary>
+        ///   The hashing algorithm.
+        /// </summary>
+        /// <value>
+        ///   Details on the hashing algorithm.
+        /// </value>
+        public HashingAlgorithm Algorithm { get; private set; }
+
+        /// <summary>
+        ///   The hashing algorithm's digest value.
+        /// </summary>
+        /// <value>
+        ///   The output of the hashing algorithm.
+        /// </value>
+        public byte[] Digest { get; private set; }
+
+        /// <summary>
+        ///   Determines if the identity hash algorithm is in use.
+        /// </summary>
+        /// <value>
+        ///   <b>true</b> if the identity hash algorithm is used; otherwise, <b>false</b>.
+        /// </value>
+        /// <remarks>
+        ///   The identity hash is used to inline a small amount of data into a <see>
+        ///       <cref>Lib.P2P.Cid</cref>
+        ///   </see>
+        ///   .
+        ///   When <b>true</b>, the <see cref="Digest"/> is also the content.
+        /// </remarks>
+        public bool IsIdentityHash => Algorithm.Code == 0;
+
+        /// <summary>
+        ///   Writes the binary representation of the multihash to the specified <see cref="Stream"/>.
+        /// </summary>
+        /// <param name="stream">
+        ///   The <see cref="Stream"/> to write to.
+        /// </param>
+        /// <remarks>
+        ///   The binary representation is a 1-byte <see cref="HashingAlgorithm.Code"/>,
+        ///   1-byte <see cref="HashingAlgorithm.DigestSize"/> followed by the <see cref="Digest"/>.
+        /// </remarks>
+        public void Write(Stream stream)
+        {
+            using (var cos = new CodedOutputStream(stream, true))
+            {
+                Write(cos);
+            }
+        }
+
+        /// <summary>
+        ///   Writes the binary representation of the multihash to the specified <see cref="Google.Protobuf.CodedOutputStream"/>.
+        /// </summary>
+        /// <param name="stream">
+        ///   The <see cref="Google.Protobuf.CodedOutputStream"/> to write to.
+        /// </param>
+        /// <remarks>
+        ///   The binary representation is a <see cref="Varint"/> of the <see cref="HashingAlgorithm.Code"/>,
+        ///   <see cref="Varint"/> of the <see cref="HashingAlgorithm.DigestSize"/> followed by the <see cref="Digest"/>.
+        /// </remarks>
+        public void Write(CodedOutputStream stream)
+        {
+            if (stream == null)
+            {
+                throw new ArgumentNullException(nameof(stream));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The parameter name '' is not declared in the argument list.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)